os/arch/arm/src/stm32l4/stm32l4_allocateheap.c

```diff
@@ -335,90 +98,39 @@ void up_allocate_heap(FAR void **heap_start, size_t *heap_size)
  *
  ****************************************************************************/
 
-#if defined(CONFIG_BUILD_PROTECTED) && defined(CONFIG_MM_KERNEL_HEAP)
 void up_allocate_kheap(FAR void **heap_start, size_t *heap_size)
 {
-  /* Get the unaligned size and position of the user-space heap.
-   * This heap begins after the user-space .bss section at an offset
-   * of CONFIG_MM_KERNEL_HEAPSIZE (subject to alignment).
-   */
-
-	uintptr_t ubase = (uintptr_t)USERSPACE->us_bssend;
-
-  /* Is ubase in SRAM1 region? */
-
-	if (ubase < SRAM1_END) {
-    /* Yes, let's register remain region as a heap. */
-
-		size_t    usize = SRAM1_END - ubase;
-		int       log2;
-
-    /* Adjust that size to account for MPU alignment requirements.
-     * NOTE that there is an implicit assumption that the SRAM1_END
-     * is aligned to the MPU requirement.
-     */
-
-		log2  = (int)mpu_log2regionfloor(usize);
-		DEBUGASSERT((SRAM1_END & ((1 << log2) - 1)) == 0);
-
-		usize = (1 << log2);
-		ubase = SRAM1_END - usize;
-
-    /* Return the kernel heap settings (i.e., the part of the heap region
-     * that was not dedicated to the user heap).
-     */
-
-		*heap_start = (FAR void *)USERSPACE->us_bssend;
-		*heap_size  = ubase - (uintptr_t)USERSPACE->us_bssend;
+	*heap_start = (FAR void *)(g_idle_topstack & ~(0x7));
```

Comment:
Originally, why we remain this is we want to use each vendor's heap allocation code.
If we want to use our generic heap allocation code, we can use up_allocateheap instead of this.
Is it possible now also?

Reply:
These changes were make to simplify the heap alloaction.
Any vendor specific setting for ram usage can be pre-calculated and set in the RAM_KREGIONx config.
In this case, we can use the common up_alocateheap instead of using the chip specific code. 
In a similar way, it is possible to move mpuinit.c, blocktask.c, unblocktask.c and other files to common folder. 
If we agree upon this, I will make the required changes in a new PR.